lib/view-transitions.js

```diff
@@ -0,0 +1,72 @@
+// Disclaimer: Importing the `store` using a global is just a temporary solution.
+const { store, directive, useContext, useLayoutEffect } =
+	window.__experimentalInteractivity;
+
+// data-wp-view-transitions-key--[option]
+directive(
+	'transition',
+	({
+		directives: { transition },
+		context,
+		evaluate,
+		element,
+		store: { state },
+	}) => {
+		const contextValue = useContext(context);
+		Object.values(transition).forEach((key) => {
+			const transitionKey = 'item-' + key;
+			useLayoutEffect(() => {
+				// If there is no Key, set it to the main option if exists.
+				if (
+					!state.core.navigation.viewTransitionKey &&
+					element.props['data-wp-transition-main']
+				) {
+					state.core.navigation.viewTransitionKey = transitionKey;
+					state.core.navigation.viewTransitionElement =
+						element.ref.current;
+				}
+
+				// If the key matches, add transition. If not, remove it.
+				if (state.core.navigation.viewTransitionKey === transitionKey) {
+					element.ref.current.style.viewTransitionName =
+						transitionKey;
+				} else {
+					element.ref.current.style.viewTransitionName = '';
+				}
+
+				return evaluate(key, { context: contextValue });
+			});
```

Comment:
We can do this to avoid unnecessary runs.

```suggestion
			}, [state.core.navigation.viewTransitionKey]);
```

Reply:
I made this change but the transition from movie to an actor stop working as expected. Although I believe it is more related to the way we are adding the `view-transition` names. I'll explore that deeper.

Reply:
Don't do this then. It's not necessary.